motionplan/constraint.go

```diff
@@ -359,6 +367,28 @@ func NewCollisionConstraint(
 	return constraint, nil
 }
 
+// NewInteractionSpaceConstraint will determine if the given list of robot geometries are encompassed by the
+// given list of interactionSpaces. An interaction space is a geometry a robot must remain within while navigating a path.
+func NewInteractionSpaceConstraint(robotGeoms, interactionSpaces []spatial.Geometry) StateConstraint {
+	// determine if we would like to also return an ik.StateMetric to measure how much a robot geometry is encompasses by a given
+	// interaction space geometry
+	f := func(state *ik.State) bool {
```

Comment:
The behavior here is that *all* robot geometries must be fully encompassed by *all* interaction spaces simultaneously. Is that the intention?

Reply:
I do not think this is the case.
What I have written, I believe, will return true iff there exists at least one interaction space which encompasses the robot's geometries. In fact, we return true on the first success case we have.
If we make it through the entire loop(s) we then know for sure that all of the defined interaction spaces have failed at encompassing the robot's geometries.

Reply:
You're correct; my apologies.